### PR TITLE
Feature/unittest velo 593 case 11 21

### PR DESCRIPTION
**Pull request by athiwatp**

### CODE REVIEW

File: contracts/modules/core/Heart.sol [hunk 1]
```diff
@@ -103,6 +103,7 @@ contract Heart is WhitelistAdminRole, IHeart {
         return collateralRatios[assetCode];
     }
 
+
```
Comment (Kafakk) [inline contracts/modules/core/Heart.sol]: remove space

File: test/sol/core/TestHeart.sol [hunk 2]
```diff
@@ -130,4 +133,131 @@ contract TestHeart {
...
+        bool expectedTrustedPartner = true;
+        heart.setTrustedPartner(address(1));
+
+        Assert.equal(heart.isTrustedPartner(address(1)), expectedTrustedPartner, "heart.setTrustedPartner() from address(1) should be true");
+    }
+
+    function testIsTrustedPartner_Fail_WhenNotInTrustedPartnerList() public {
+        bool expectedTrustedPartner = false;
+
+        Assert.equal(heart.isTrustedPartner(address(2)), expectedTrustedPartner, "heart.setTrustedPartner() from address(2) should be false");
```
Comment (Kafakk) [inline test/sol/core/TestHeart.sol]:
```suggestion
        Assert.equal(heart.isTrustedPartner(address(2)), expectedTrustedPartner, "heart. isTrustedPartner() from address(2) should be false");
```

File: test/sol/core/TestHeart.sol [hunk 3]
```diff
@@ -130,4 +133,131 @@ contract TestHeart {
...
+        heart.setDrsAddress(address(this));
+        heart.setCollateralAsset(expectedCollateralAsset, ierc20Address, ratio);
+        heart.setCollateralRatio(expectedCollateralAsset, ratio);
+        heart.collectFee(expectedCollectedFee, expectedCollateralAsset);
+        Assert.equal(heart.getCollectedFee(expectedCollateralAsset), expectedCollectedFee, "heart.getCollectedFee() should be collectedFee 20");
+
+        heart.withdrawFee(expectedCollateralAsset, expectedAmount);
+        uint256 balanceCollectedFee = heart.getCollectedFee(expectedCollateralAsset);
+
+        Assert.equal(balanceCollectedFee, expectedBalanceCollectedFee, "heart.getCollectedFee() should be collectedFee 19");
```
Comment (Kafakk) [inline test/sol/core/TestHeart.sol]:
```suggestion
        Assert.equal(balanceCollectedFee, expectedBalanceCollectedFee, "heart. withdrawFee() should withdraw fee correctly");
```

File: test/sol/core/TestHeart.sol [hunk 4]
```diff
@@ -96,38 +99,160 @@ contract TestHeart {
...
+        heart.getCreditIssuanceFee();
+
+        Assert.equal(heart.getCreditIssuanceFee(), expectedCreditIssuanceFee, "heart.getCreditIssuanceFee() should be 1");
+    }
+
+    function testSetTrustedPartner_Success() public {
+        bool expectedTrustedPartner = true;
+        heart.setTrustedPartner(address(1));
+
+        Assert.equal(heart.isTrustedPartner(address(1)), expectedTrustedPartner, "heart.setTrustedPartner() from address(1) should be true");
```
Comment (jura-b) [inline test/sol/core/TestHeart.sol]: Assert.isTrue

Comment (jura-b) [reply test/sol/core/TestHeart.sol]:
```   
 function testSetTrustedPartner_Success() public {
        Assert.isFalse(heart.isTrustedPartner(address(1)));
        heart.setTrustedPartner(address(1));
        Assert.isTrue(heart.isTrustedPartner(address(1)));
    }

File: test/sol/core/TestHeart.sol [hunk 5]
```diff
@@ -56,14 +59,14 @@ contract TestHeart {
         Assert.equal(uint(reserveFreezeSecond), uint(newSeconds), "heart.getReserveFreeze() should get reserve freeze seconds properly");
     }
 
-    function testSetDrsAddress() public {
+    function testSetDrsAddress_Success() public {
         address expected = 0xBe40F54aDc30b5e2d2b7CEAc2Fc600e786c49B81;
```
Comment (jura-b) [inline test/sol/core/TestHeart.sol]: address expected = address(10);

File: test/sol/core/TestHeart.sol [hunk 6]
```diff
@@ -56,14 +59,14 @@ contract TestHeart {
...
         heart.setDrsAddress(expected);
 
         Assert.equal(heart.getDrsAddress(), expected, "heart.setDrsAddress() should set drs address properly");
     }
 
-    function testGetDrsAddress() public {
+    function testGetDrsAddress_Success() public {
         address firstAddress = 0xBe40F54aDc30b5e2d2b7CEAc2Fc600e786c49B81;
```
Comment (jura-b) [inline test/sol/core/TestHeart.sol]:
```
address expected = address(10);
address expected = address(11);

File: test/sol/core/TestHeart.sol [hunk 7]
```diff
@@ -96,38 +99,160 @@ contract TestHeart {
...
+    }
+
+    function testWithdrawFee_Success() public {
+        uint256 expectedCollectedFee = 20;
+        uint256 expectedAmount = 1;
+        uint256 expectedBalanceCollectedFee = 19;
+        bytes32 expectedCollateralAsset = 0x56454c4f00000000000000000000000000000000000000000000000000000000;
+        // VELO
+
+        address ierc20Address = address(mockIERC20);
```
Comment (jura-b) [inline test/sol/core/TestHeart.sol]: mockIERC20Address

File: test/sol/core/TestHeart.sol [hunk 8]
```diff
@@ -96,38 +99,160 @@ contract TestHeart {
...
+
+        Assert.isFalse(result, "heart.collectFee() should throw error only DRSSC can update the collected fee");
+    }
+
+    function testGetCollectedFee_Success() public {
+        uint256 expectedFee = 1;
+        bytes32 expectedCollateralAsset = 0x56454c4f00000000000000000000000000000000000000000000000000000000;
+        // VELO
+
+        heart.setDrsAddress(address(this));
```
Comment (jura-b) [inline test/sol/core/TestHeart.sol:208]: remove this

**Pull request merged**